sklearn/preprocessing/data.py

```diff
@@ -2718,23 +2718,18 @@ def _box_cox_inverse_tranform(self, x, lmbda):
     def _yeo_johnson_inverse_transform(self, x, lmbda):
         """Return inverse-transformed input x following Yeo-Johnson inverse
         transform with parameter lambda.
-
-        Notes
-        -----
-        We're comparing lmbda to 1e-19 instead of strict equality to 0. See
-        scipy/special/_boxcox.pxd for a rationale behind this
         """
-        x_inv = np.zeros(x.shape, dtype=x.dtype)
+        x_inv = np.zeros_like(x)
         pos = x >= 0
 
         # when x >= 0
-        if lmbda < 1e-19:
+        if abs(lmbda) < np.spacing(1.):
```

Comment:
ok so the ``abs`` here is the bugfix, right, and the rest is minor stability changes?

Reply:
yes